cheroot/server.py

```diff
@@ -1066,50 +749,72 @@ def respond(self):
                         'maximum allowed bytes.',
                     )
                 return
-            self.rfile = KnownLengthRFile(self.conn.rfile, cl)
+            self.rfile = KnownLengthRFile(
+                self.h_conn,
+                self.conn.rfile, self.conn.wfile, cl,
+            )
 
-        self.server.gateway(self).respond()
-        self.ready and self.ensure_headers_sent()
+        if self.h_conn.client_is_waiting_for_100_continue:
+            mini_headers = ()
+            go_ahead = h11.InformationalResponse(
+                status_code=100,
+                headers=mini_headers,
+            )
+            bytes_out = self.h_conn.send(go_ahead)
+            self.conn.wfile.write(bytes_out)
+        try:
+            self.server.gateway(self).respond()
+            self.ready and self.ensure_headers_sent()
+        except errors.MaxSizeExceeded as e:
```

Comment:
Normally, you need to have only one statement in the try block. Otherwise, it may be harmful when the exception starts happening in a different place.